tests/RequestListenerTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Opstalent\CrudBundle\Tests;
+
+use Opstalent\CrudBundle\CrudHandlingInterface;
+use Opstalent\CrudBundle\RequestListener;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Opstalent\CrudBundle\FormConfigResolver;
+use Opstalent\CrudBundle\FormFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Opstalent\CrudBundle\Tests\Entity\EntityWithAnnotation;
+
+/**
+ * Class RequestListenerTest
+ * @author Szymon Kunowski <szymon.kunowski@gmail.com>
+ * @package Opstalent\CrudBundle
+ */
+class RequestListenerTest extends TestCase
+{
+    /**
+     * @param bool $master
+     * @param Request $request
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    public function getResponseEventMock(bool $master = false, Request $request)
+    {
+        $mock = $this
+            ->getMockBuilder(GetResponseEvent::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $mock->method('isMasterRequest')->willReturn($master);
+        $mock->method('getRequest')->willReturn($request);
+        return $mock;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    public function getFormConfigResolverMock()
+    {
+        return $this
+            ->getMockBuilder(FormConfigResolver::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    public function getFormFactoryMock()
+    {
+        return $this
+            ->getMockBuilder(FormFactory::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    public function getRequestMock()
+    {
+        $requestMock =  $this
+            ->getMockBuilder(Request::class)
+            ->setConstructorArgs([[], [], [], [], [], [], null])
+            ->getMock();
+
+        $requestMock->query = new ParameterBag();
+        $requestMock->request = new ParameterBag();
+        $requestMock->attributes = new ParameterBag();
+        $requestMock->cookies = new ParameterBag();
+        $requestMock->files = new ParameterBag();
+        $requestMock->server = new ParameterBag();
+        $requestMock->headers = new ParameterBag();
```

Comment:
do you need all these bags?

Reply:
sadly yes

Reply:
mock is not working without this? or what?

Reply:
there is function that clone every this params (not in Request) and if i don't init like this this will show error i could investigate more.